front/src/applications/operationalStudies/components/MacroEditor/types.ts

```diff
@@ -153,3 +159,20 @@ export type NetzgrafikDto = {
     filterSettings: unknown[];
   };
 };
+
+type NGECreateEvent = {
+  type: 'create';
+  trainrunSection: OperationTrainrunSection;
+};
+
+export type NGEDeleteEvent = {
+  type: 'delete';
+  trainrun: Trainrun;
```

Comment:
nit: you can only give the trainrunId to delete
```suggestion
  trainrunId: number;
```

Reply:
We can't change this without changing NGE. These data structures describe values coming from NGE.

The full trainrun isn't useful to us, but might be to other NGE users.

Reply:
OK got it, it's fine for me :+1: